src/server/saved_objects/service/create_saved_objects_service.js

```diff
@@ -32,34 +31,48 @@ import { SavedObjectsClient } from './saved_objects_client';
 // remove the migrator from the saved objects client and leave only document validation here.
 export function createSavedObjectsService(server, schema, serializer, migrator) {
   const mappings = server.getKibanaIndexMappingsDsl();
-  const repositoryProvider = new SavedObjectsRepositoryProvider({
-    index: server.config().get('kibana.index'),
-    migrator,
-    mappings,
-    schema,
-    serializer,
-  });
+  const allTypes = Object.keys(getRootPropertiesObjects(mappings));
+  const visibleTypes = allTypes.filter(type => !schema.isHiddenType(type));
+
+  const createRepository = (callCluster, extraTypes = []) => {
```

Comment:
Could we add some tests for this? That's one of the reasoning for initially introducing the dedicated `SavedObjectsRepositoryProvider` which has been removed, it made testing this easier

Reply:
I will rework these tests to be useful